### DISCUSSION TITLE
Can frankenphp load different PHP versions dynamically?

### DISCUSSION THREAD

**Discussion by shyim** [php/frankenphp#701] [Q&A]
I compiled Frankenphp with 8.3, can I use now libphp.so from 8.2? Or should I always compile frankenphp with the proper major version?

fun fact 8.3 compiled, does not work with 8.2 as this symbol is missing `frankenphp: symbol lookup error: frankenphp: undefined symbol: php_tsrm_startup_ex`

`frankenphp php-cli <some-php-file>` does work with both PHP versions

**Comment by dunglas** [comment]
This is likely because of https://github.com/dunglas/frankenphp/pull/655.

I'm not sure if it's possible to workaround this at runtime.

**Reply by shyim** [reply]
It's OK if that does not work. maybe document it. I have no problem to compile it for each PHP major 👍

**Reply by dunglas** [reply]
That would be cool to be able to swap `libphp.so` without having to recompile.

**Reply by shyim** [reply]
Should I create an issue from this?
